MapboxSearch/autofill/src/main/java/com/mapbox/search/autofill/AddressAutofill.kt

```diff
@@ -62,17 +62,20 @@ public interface AddressAutofill {
          * By default [LocationProvider] is provided by [LocationServiceFactory].
          * Note that this class requires [Manifest.permission.ACCESS_COARSE_LOCATION] or
          * [Manifest.permission.ACCESS_FINE_LOCATION] to work properly.
+         * @param locationObservationTimeout to be used in LocationEngineAdapter. null - default value
          *
          * @return a new instance instance of [AddressAutofill].
          */
         @JvmStatic
         @JvmOverloads
         public fun create(
             locationProvider: LocationProvider? = defaultLocationProvider(),
+            locationObservationTimeout: Long? = null
```

Comment:
As I [mentioned](https://github.com/mapbox/mapbox-search-android/pull/199/#issuecomment-2002173076) in the previous PR comment, it's preferable to address the root cause rather than attempting workarounds that impact the public API.

The `locationObservationTimeout` parameter seems unclear in its relation to the AddressAutofill API.

Furthermore, it's error-prone; if multiple instances of `AddressAutofill` are created with different `locationObservationTimeout` values, only the latest one will be utilized.

If addressing the root cause isn't feasible, there're alternative solutions that don't impact the C++ code. For instance, accessing the last known location each time a new search request is initiated.

Reply:
I should've put it into the description (although I mentioned it verbally in our team meeting) but this PR demonstrates two ways to set the timeout value: through setting a global variable and through passing a parameter in constructors. I favor the first method (less code) but provided the second method as an option. We aim to decide on and leave only one of them. Apparently, the global variable is the least disliked option.

Speaking of the last known location: it's provided by an asynchronous method, so this whole adapter is only needed to convert an asynchronous call to the eventual result value. We return the last know location already if it's not stale enough to warrant an update.